pkg/block/fetcher.go

```diff
@@ -540,6 +574,34 @@ func contains(s1 []ulid.ULID, s2 []ulid.ULID) bool {
 	return true
 }
 
+// ReplicaLabelRemover is a MetaFetcher modifier modifies external labels of existing blocks, it removes given replica labels from the metadata of blocks that have it.
+type ReplicaLabelRemover struct {
+	logger log.Logger
+
+	replicaLabels []string
+}
+
+// NewReplicaLabelRemover creates a ReplicaLabelRemover.
+func NewReplicaLabelRemover(logger log.Logger, replicaLabels []string) *ReplicaLabelRemover {
+	return &ReplicaLabelRemover{logger: logger, replicaLabels: replicaLabels}
+}
+
+// Modify modifies external labels of existing blocks, it removes given replica labels from the metadata of blocks that have it.
+func (r *ReplicaLabelRemover) Modify(_ context.Context, metas map[ulid.ULID]*metadata.Meta, metrics *syncMetrics, view bool) error {
```

Comment:
so... why we pass syncMetrics if we modify here? We could pass `modified` counter! It would be much cleaner, right?

Reply:
So we're fetching the whole metrics struct name is misleading. I'll change the name, it should be `fetcherMetrics`.

Reply:
Still this is not addressed I meant something else. Why we need to pass whole `fetcherMetrics` struct when we need only on metric? It makes this interface more broad than it should. (: WDYT?

Reply:
These functions are used in https://github.com/thanos-io/thanos/blob/594fed94846ed9caedffd66dacdef8446c8b7582/pkg/block/fetcher.go#L354

I didn't want to introduce another interface for modifiers since we only have one right now. 
If we decide to go down that path I can introduce a dedicated interface and distinguish between counters. 

Or I can change the interface and check in this loop and if it's a modifier and pass another metric.

I chose this because I wanted to minimize changes, design changes.